in_toto/attestation/common/common.go

```diff
@@ -0,0 +1,29 @@
+package common
+
+import (
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+/*
+PredicatePbToStruct converts a given protobuf representation of
+an in-toto attestation Predicate into a generic Struct struct
+needed to construct an in-toto Attestation Framework v1
+compliant Statement. If there are any marshalling problems,
+this function returns an error.
+*/
+func PredicatePbToStruct(predicatePb proto.Message) (*structpb.Struct, error) {
```

Comment:
I wonder if want this to be `internal/`, I don't think we want to accidentally have people use this. Or it could be a private function in `attestation`.

Reply:
I think I had set it public since it's called from within `runlib`, but either moving this to `internal/` or even as a private function to be called only within `GenerateValidStatement()` could work. We would have to change the signature of `GenerateValidStatement()` to take a protobuf message instead of Struct in the latter case, not sure how I feel about that.